add stable window documentatino to "scale-bounds" (#4514)
knative/docs: docs/serving/autoscaling/scale-bounds.md

          allow-zero-initial-scale: "true"
    ```

> ### Scale Down Delay

Reviewer: ```suggestion
## Scale Down Delay
```

Author: :+1:

Scale Down Delay specifies a time window which must pass at reduced concurrency
before a scale-down decision is applied. This can be useful, for example, to
keep containers around for a configurable duration to avoid a cold start
penalty if new requests come in. Unlike setting a lower bound, the revision
will eventually be scaled down if reduced concurrency is maintained for the

> delay period. *only supported for KPA*

Reviewer 2: Maybe convert this to something like:

`!!! note "Only Supported for the default KPA autoscaler class"`

(so we get nice formatting)

> delay period. !!! note "Only Supported for the default KPA autoscaler class"

Reviewer: ```suggestion
delay period. 
!!! note 
    Only supported for the default KPA autoscaler class.
```

Author: :+1:

          allow-zero-initial-scale: "true"
    ```

## Stable window

> The stable window defines minimal time that will pass between descisions to change the scale of a revision.

Reviewer 2: Is this true? What if we're in panic mode?

Author: you are right, I will add reference to panic mode.

          scale-down-delay: "15m"
    ```

## Stable window

> The stable window defines minimal time that passes between decisions to change the scale of a revision. [Panic mode](kpa-specific.md) can override this setting.

Reviewer 2: I think this isn't quite right. We average over the stable and panic windows to get the _input_ for decisions, but we actually make decisions [every two seconds by default](https://github.com/knative/serving/blob/61178b2e8bf8d8afe870151ff59ee9724739b496/pkg/autoscaler/scaling/multiscaler.go#L36) . I'd say something like:

```suggestion
The stable window defines the sliding time window over which metrics are averaged to provide the input for scaling decisions when the autoscaler is not in [Panic mode](kpa-specific.md).
```

> The stable window defines minimal time that will pass between descisions to change the scale of a revision. [Panic mode](kpa-specific.md) can override this setting.

Reviewer: ```suggestion
The stable window defines minimal time that passes between decisions to change the scale of a revision. [Panic mode](kpa-specific.md) can override this setting.
```

Author: :+1:

* **Global key:** `stable-window`
* **Per-revision annotation key:** `autoscaling.knative.dev/window`
* **Possible values:** Duration, `6s` <= value <= `1h`
* **Default:** `60s`

!!! note

>     During scale down, the last Replica will only be removed after there has not been any traffic to the Revision for the entire duration of the stable window.

Reviewer 2: I _think_ this is true, in that if there hasn't been any traffic during the stable window the average metric will not be 0, so we won't scale down, but I'm not completely sure that we guarantee this e.g. if [ExponentialWeightedBuckets](https://github.com/knative/serving/blob/main/pkg/autoscaler/aggregation/bucketing.go#L150) were used I wonder if the average might go to zero faster. I'm probably being too pedantic here, though 😅.

Reviewer: @julz Do you think slightly changing the wording here would make sense - maybe to something less definite like:
> During scale down, the last Replica is usually removed after there has not been any traffic to the Revision for the entire duration of the stable window

Or do you think what is here now is OK?

Reviewer 2: sorry for slow response @snneji I think I might've replied to this on mobile and forgot to hit submit 🤦; yeah I think that looks good

        autoscaler:
          scale-down-delay: "15m"
    ```
## Stable window

The stable window defines the sliding time window over which metrics are averaged to provide the input for scaling decisions when the autoscaler is not in [Panic mode](kpa-specific.md).

* **Global key:** `stable-window`
* **Per-revision annotation key:** `autoscaling.knative.dev/window`
* **Possible values:** Duration, `6s` <= value <= `1h`
* **Default:** `60s`

!!! note

>     During scale down, the last Replica will only be removed after there has not been any traffic to the Revision for the entire duration of the stable window.

Reviewer: ```suggestion
    During scale down, in most cases the last Replica is removed after there has been no traffic to the Revision for the entire duration of the stable window.
```

**Example:**

=== "Per Revision"
    ```yaml
    apiVersion: serving.knative.dev/v1
    kind: Service
    metadata:
      name: helloworld-go
      namespace: default
    spec:
      template:
        metadata:
          annotations:

>             autoscaling.knative.dev/window: "40s"

Reviewer 3: I think the annotation is on the Service Resource not on the pod spec right?

Reviewer: @csantanapr do you have a suggested change that should be made here? I'd like to get this PR merged.

Reviewer 3: >I think the annotation is on the Service Resource not on the pod spec right?

I double-check, I don't know what I was thinking when I wrote the comment

The annotation is in the correct space under `spec.metadata.annotations` 
Ignore my comment, the way you have it is correct.
